src/context/TransactionsProvider/TransactionsProvider.tsx

```diff
@@ -42,7 +46,19 @@ export const TransactionsProvider = ({ children }: TransactionsProviderProps): J
             await adapter.subscribeTxs(
               { wallet, accountType, ...accountParams },
               msg => {
-                dispatch(txHistory.actions.onMessage({ message: { ...msg, accountType } }))
+                const accountSpecifierObj = accountSpecifiers.reduce((acc, cur) => {
+                  if (acc) return acc
+                  const [k, v] = Object.entries(cur)[0]
+                  if (k === caip2) return v.toLowerCase()
+                  return acc
+                }, '')
```

Comment:
this is kinda weird... can you add a comment explaining what it's doing?

Reply:
this is a good point, it's going to work for eth but not btc correctly. 

we actually need a mapping here between accountType and x/y/z pub here; legacy is xpub, segwit is ypub, segwit native is zpub, so it should search for those accordingly.